PocketKit/Sources/Sync/PocketSource.swift

```diff
@@ -303,6 +303,33 @@ extension PocketSource {
         )
         enqueue(operation: operation, task: .save(localID: item.objectID.uriRepresentation(), url: url))
     }
+    
+    public func addTags(item: SavedItem, tags: [String]) {
+        guard let remoteID = item.remoteID else {
+            return
+        }
+        
+        tags.compactMap { $0 }.forEach({ tag in
+            let fetchRequest = Requests.fetchTag(byName: tag)
+            fetchRequest.fetchLimit = 1
+            let fetchedTag = try? space.context.fetch(fetchRequest).first ?? Tag(context: space.context)
+            guard let fetchedTag = fetchedTag else { return }
+            fetchedTag.name = tag
```

Comment:
I could see moving this logic into `Space` by adding a method `fetchOrCreateByName(name: String)`. Would help keep this method here a bit more focused on building up the mutations.